src/pupil_labs/neon_recording/timeseries/blinks.py

```diff
@@ -2,49 +2,50 @@
 from typing import TYPE_CHECKING
 
 import numpy as np
+import numpy.typing as npt
 
-from pupil_labs.neon_recording.stream.array_record import (
+from pupil_labs.neon_recording.timeseries.array_record import (
     Array,
     Record,
     fields,
 )
+from pupil_labs.neon_recording.timeseries.timeseries import Timeseries, TimeseriesProps
 from pupil_labs.neon_recording.utils import (
     find_sorted_multipart_files,
     load_multipart_data_time_pairs,
 )
 
-from .stream import Stream, StreamProps
-
 log = logging.getLogger(__name__)
 
 if TYPE_CHECKING:
     from ..neon_recording import NeonRecording
 
 
-class BlinkProps(StreamProps):
-    start_ts = fields[np.int64]("start_timestamp_ns")
+class BlinkProps(TimeseriesProps):
+    start_ts: npt.NDArray[np.int64] = fields[np.int64]("start_timestamp_ns")  # type:ignore
```

Comment:
This annotation is incorrect, start_ts can be both `npt.NDArray[np.int64]` or `np.int64` depending on if the Props class is being used in a Timeseries or Record context.

Reply:
True, that is the case for all the Props' type annotations I added. Previously, we did not have any type inference for those attributes, which are just inferred as `unknown`. Do you have an idea how to do this properly?

Reply:
It was a different issue, fixed in https://github.com/pupil-labs/pl-neon-recording/pull/36/commits/42a3c24f7d5a3d7221e28c3654e6f686a82bb8e4